src/app/plugins/shadowban.plugin.ts

```diff
@@ -0,0 +1,94 @@
+import { Plugin } from '../../common/plugin';
+import { IContainer, IMessage, ChannelType } from '../../common/types';
+import Constants from '../../common/constants';
+import { CategoryChannel, Collection, User } from 'discord.js';
+
+export class ShadowBanPlugin extends Plugin {
+  public name: string = 'Shadowban Plugin';
+  public description: string = 'Disables a users ability to view public channels.';
+  public usage: string = 'shadowban <ban|unban> <user>';
+  public pluginAlias = [];
+  public permission: ChannelType = ChannelType.Staff;
+  public pluginChannelName: string = Constants.Channels.Staff.UserOffenses;
+
+  private BANNED_CATEGORIES: string[] = [
+    'GENERAL & SCHOOL LIFE',
+    'DAILY ROUTINE',
+    'HELP',
+    'SPECIAL TOPICS',
+    'MISCELLANEOUS',
+    'AUDIO CHANNELS',
+  ];
+
+  constructor(public container: IContainer) {
+    super();
+  }
+
+  public validate(message: IMessage, args: string[]) {
+    return args && args.length > 1;
+  }
+
+  public async execute(message: IMessage, args?: string[]) {
+    if (!args) {
+      return;
+    }
```

Comment:
should be able to skip this by making the param not `?` and `validate` is always called before execute